src/generator/AutoRest.Go/GoCodeGenerator.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+using AutoRest.Core;
+using AutoRest.Core.ClientModel;
+using AutoRest.Go.TemplateModels;
+using AutoRest.Go.Templates;
+
+namespace AutoRest.Go
+{
+    public class GoCodeGenerator : CodeGenerator
+    {
+        private readonly GoCodeNamer _namingFramework;
+
+        public GoCodeGenerator(Settings settings) : base(settings)
+        {
+            _namingFramework = new GoCodeNamer();
+        }
+
+        public override string Name
+        {
+            get { return "Go"; }
+        }
+
+        public override string Description
+        {
+            // TODO resource string.
+            get { return "Go Client Libraries"; }
+        }
+
+        public override string UsageInstructions
+        {
+            get { return ""; }
+        }
+
+        public override string ImplementationFileExtension
+        {
+            get { return ".go"; }
+        }
+
+        /// <summary>
+        /// Normalizes client model by updating names and types to be language specific.
+        /// </summary>
+        /// <param name="serviceClientModel"></param>
+        public override void NormalizeClientModel(ServiceClient serviceClientModel)
+        {
+            // Add the current package name as a reserved keyword
+            _namingFramework.ReserveNamespace(Settings.Namespace);
+
+            PopulateAdditionalProperties(serviceClientModel);
+            
+            _namingFramework.NormalizeClientModel(serviceClientModel);
+            _namingFramework.ResolveNameCollisions(serviceClientModel, Settings.Namespace,
+                Settings.Namespace + ".Models");
+        }
+
+        private void PopulateAdditionalProperties(ServiceClient serviceClientModel)
```

Comment:
>PopulateAdditionalProperties [](start = 21, length = 28)

we should remove this if not used or provide more comments why is it commented out.